staticaddr/deposit/fsm.go

```diff
@@ -274,6 +324,23 @@ func (f *FSM) updateDeposit(notification fsm.Notification) {
 	}
 }
 
+// Don't update the deposit if we are in an initial state or if we are
+// transitioning from an initial state to a failed state.
```

Comment:
I propose to move this comment inside the function and add godoc starting with the name of the function.